skimage/feature/blob.py

```diff
@@ -53,21 +55,28 @@ def _blob_overlap(blob1, blob2):
     if d <= abs(r1 - r2):
         return 1
 
-    ratio1 = (d ** 2 + r1 ** 2 - r2 ** 2) / (2 * d * r1)
-    ratio1 = np.clip(ratio1, -1, 1)
-    acos1 = arccos(ratio1)
+    if n_dim == 2:
+        ratio1 = (d ** 2 + r1 ** 2 - r2 ** 2) / (2 * d * r1)
+        ratio1 = np.clip(ratio1, -1, 1)
+        acos1 = arccos(ratio1)
+
+        ratio2 = (d ** 2 + r2 ** 2 - r1 ** 2) / (2 * d * r2)
+        ratio2 = np.clip(ratio2, -1, 1)
+        acos2 = arccos(ratio2)
 
-    ratio2 = (d ** 2 + r2 ** 2 - r1 ** 2) / (2 * d * r2)
-    ratio2 = np.clip(ratio2, -1, 1)
-    acos2 = arccos(ratio2)
+        a = -d + r2 + r1
+        b = d - r2 + r1
+        c = d + r2 - r1
+        d = d + r2 + r1
+        area = r1 ** 2 * acos1 + r2 ** 2 * acos2 \
+              - 0.5 * sqrt(abs(a * b * c * d))
```

Comment:
PEP8 suggests parens for line continuations, and linebreaks *after* binary operators (with some further tweaks from me):

```python
area = (acos1 * r1**2 + acos2 * r2**2 -
        0.5 * sqrt(abs(a * b * c * d)))
```